### PR TITLE
Eager mode: cuda kernel support

### PR DESCRIPTION
**Pull request by souptc**
Deprecate the legacy CudaContext and replace the kernel using the new interface, which can be invoked from eager mode.

### CODE REVIEW

File: test/shared_test/test_exceptions.cc [hunk 1]
```diff
@@ -87,7 +87,7 @@ TEST(Exceptions, TestApiTryCatch_ThrowInModelLoad) {
   EXPECT_EXIT(fail_fn(), ::testing::KilledBySignal(SIGABRT), ".*");
 #endif
 #else
-  // ORT catches the exceptions thrown by the custom op and rethrows them as Ort::Exception
```
Comment (wenbingl) [inline test/shared_test/test_exceptions.cc:90]: Is it done by any C++ formatter tool? it looks like unnecessary change.

**Review by wenbingl — APPROVED** [review]

File: test/static_test/test_cuda_eager.cc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+public:
+  void* Alloc(size_t size) override { 
+    void* p = nullptr;
+    cudaMalloc((void**)&p, size);
+    return p;
+  }
+  void Free(void* p) override { cudaFree(p); }
+};
+
+TEST(CudaOp, test_eager_negpos) {
```
[the remark this answers is not in the document]
Comment (souptc) [reply test/static_test/test_cuda_eager.cc:41]: I didn't get it, you mean a build option?
